src/System.Collections/tests/Generic/Comparers/Comparer.Tests.cs

```diff
@@ -0,0 +1,321 @@
+            }
+        }
+
+        public static IEnumerable<object[]> IntEnumComparisonsData()
+        {
+            var testCases = new[]
+            {
+                Tuple.Create(-1, 4, -1),
+                Tuple.Create(-222, -375, 1),
+                Tuple.Create(int.MinValue, int.MinValue, 0),
+                // The same principle applies for overflow in signed types as above,
+                // the implementation should not cast to an unsigned type
+                Tuple.Create(int.MaxValue, int.MinValue, 1),
+                Tuple.Create(int.MinValue + 1, int.MinValue, 1)
+            };
+
+            foreach (var testCase in testCases)
+            {
+                yield return new object[] { (IntEnum)testCase.Item1, (IntEnum)testCase.Item2, testCase.Item3 };
+                yield return new object[] { (IntEnum)testCase.Item2, (IntEnum)testCase.Item1, -testCase.Item3 };
+            }
+        }
+
+        public static IEnumerable<object[]> UIntEnumComparisonsData()
+        {
+            var testCases = new[]
+            {
+                Tuple.Create(5u, 5u, 0),
+                Tuple.Create(445u, 123u, 1),
+                Tuple.Create(uint.MaxValue, 111u, 1),
+                Tuple.Create(uint.MaxValue - 333, uint.MaxValue, -1)
+            };
+
+            foreach (var testCase in testCases)
+            {
+                yield return new object[] { (UIntEnum)testCase.Item1, (UIntEnum)testCase.Item2, testCase.Item3 };
+                yield return new object[] { (UIntEnum)testCase.Item2, (UIntEnum)testCase.Item1, -testCase.Item3 };
+            }
+        }
+
+        public static IEnumerable<object[]> LongEnumComparisonsData()
+        {
+            var testCases = new[]
+            {
+                Tuple.Create(555L, 555L, 0),
+                Tuple.Create(182912398L, 33L, 1),
+                Tuple.Create(long.MinValue, long.MaxValue, -1),
+                Tuple.Create(long.MinValue + 9, long.MinValue, 1),
+                Tuple.Create(-1L, -1L, 0)
+            };
+            
+            foreach (var testCase in testCases)
+            {
+                yield return new object[] { (LongEnum)testCase.Item1, (LongEnum)testCase.Item2, testCase.Item3 };
+                yield return new object[] { (LongEnum)testCase.Item2, (LongEnum)testCase.Item1, -testCase.Item3 };
+            }
+        }
+
+        public static IEnumerable<object[]> PlainObjectComparisonsData()
+        {
+            var obj = new object(); // this needs to be cached into a local so we can pass the same ref in twice
+
+            var testCases = new[]
+            {
+                Tuple.Create(default(object), default(object), 0),
+                Tuple.Create(obj, obj, 0), // even if it doesn't implement IComparable, if 2 refs are the same then the result should be 0
+                Tuple.Create(default(object), obj, -1) // even if it doesn't implement IComparable, if one side is null -1 or 1 should be returned
```

Comment:
`null` is necessary for the `Tuple.Create` type inference to work, otherwise it can't determine the type of `T1`.